apps/web/components/Project/Header/DefaultProjectHeader.tsx

```diff
@@ -31,45 +33,63 @@ export const DefaultProjectHeader = ({
 
   const textBackgroundColor = getTextBackgroundColor(projectPage)
 
+  const textRef = useRef<HTMLDivElement | null>(null)
+
+  const { width } = useWindowSize()
+
+  const isBelowLarge = width < theme.breakpoints.lg
+
+  const maxImageHeight = !isBelowLarge
+    ? textRef.current?.getBoundingClientRect()?.height ?? undefined
+    : undefined
+
```

Comment:
Consider not using javascript for style changes.
If I'm not mistaken, computed javascript style changes can create layout thrashing.

Is it possible to do this with CSS (maybe breakpoints)?
Perhaps @barabrian is better suited to comment on this than I am.

Reply:
This was the only way that I could make the height of the container follow the text on the left.
There are two columns here and I want to allow the left one to control the height of both. I was having a lot of issues trying do to that in css so I went for a Javascript solution.

Reply:
If the right column only contains an image, it should be possible to position the image absolute inside the right column so it wont take any space (if thats the issue, I haven't really gone through the code only based on the screenshots). But regarding this code I would strongly suggest never to use getBoundingClientRect inside `render`, it is quite expensive performance wise. I would at least suggest wrapping it in a useCallback
ex.
```js
const [height, setHeight] = useState(0);

const measuredRef = useCallback(node => {
  if (node !== null) {
    setHeight(node.getBoundingClientRect().height);
  }
}, [])
```
Then you would use `measuredRef` instead of `textRef`

But the best solution would still be css only if possible 😄

Reply:
Yep, I'll have to have another go at this. Thanks for the advice guys :)